homeassistant/components/assist_pipeline/vad.py

```diff
@@ -239,3 +258,40 @@ def _process_chunk(self, chunk: bytes) -> bool:
             )
 
         return True
+
+
+def chunk_samples(
+    samples: bytes,
+    bytes_per_chunk: int,
+    sample_buffer: AudioBuffer,
+) -> Iterable[bytes]:
+    """Yield fixed-sized chunks from samples, keeping leftover bytes in a buffer."""
+    samples_offset = 0
+    num_chunks_in_samples = len(samples) // bytes_per_chunk
+    num_bytes_left = len(samples) % bytes_per_chunk
+
+    if len(sample_buffer) > 0:
+        # Add to partial chunk in buffer
+        bytes_to_copy = min(num_bytes_left, bytes_per_chunk - len(sample_buffer))
```

Comment:
Why do you limit this to `num_bytes_left` ? Wouldn't you just want the first bytes from `samples`, even if it's larger than `bytes_left` ? 

This makes more sense I think:

```suggestion
        bytes_to_copy = min(len(samples), bytes_per_chunk - len(sample_buffer))
```